### PR TITLE
Port S3 Download Demo from C-SDK

### PR DESCRIPTION
**Pull request by sukhmanm**
Adding demo to use script-generated pre-signed URLs to establish a TLS connection and _synchronously_ download a file from an S3 bucket with multiple GET requests.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+    /* The network context for the transport layer interface. */
+    NetworkContext_t xNetworkContext = { 0 };
+    BaseType_t xIsConnectionEstablished = pdFALSE;
+    /* HTTPS Client library return status. */
+    HTTPStatus_t xHTTPStatus = HTTPSuccess;
+
+    /* Upon return, pdPASS will indicate a successful demo execution.
+    * pdFAIL will indicate some failures occurred during execution. The
+    * user of this demo must check the logs for any failure codes. */
+    BaseType_t xDemoStatus = pdPASS;
```
Comment (sarenameas) [inline FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c:276]:
```suggestion
    /*  The user of this demo must check the logs for any failure codes. */
    BaseType_t xDemoStatus = pdPASS;
```

We don't return anything in this repo

**Review by archigup — DISMISSED** [review]

**Review by sarenameas — DISMISSED** [review]

**Review by sarenameas — DISMISSED** [review]

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/mbedtls_config.h [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  **********
+ *
+ *  **********
+ *  GNU General Public License v2.0 or later:
```
Comment (cobusve) [inline FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/mbedtls_config.h:26]: GPL is a problem

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c [hunk 3]
```diff
@@ -250,17 +264,6 @@ void vStartSimpleHTTPDemo( void )
 
 /**
  * @brief Entry point of the demo.
- *
```
Comment (sarenameas) [inline FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c]: Why delete this?

Comment (sukhmanm) [reply FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c]: I had been referring to the MQTT multitask demo and saw that there was no description, but I now realize the other demos have included it so I have reverted the change. Thanks.

**Review by sarenameas — APPROVED** [review]

**Review by cobusve — APPROVED** [review]

**Pull request merged**